command/loadbot/execution.go

```diff
@@ -64,10 +66,29 @@ type metadata struct {
 	blockNumber uint64
 }
 
+type GasMetrics struct {
+	GasUsed     uint64
+	GasLimit    uint64
+	Utilization float64
+}
+
+type BlockGasMetrics struct {
+	Blocks        map[uint64]GasMetrics
+	BlockGasMutex *sync.Mutex
+}
+
 type Metrics struct {
 	TotalTransactionsSentCount uint64
 	FailedTransactionsCount    uint64
 	TransactionDuration        ExecDuration
+
+	// contracts
+	FailedContractTransactionsCount uint64
+	ContractDeploymentDuration      ExecDuration
+	ContractAddress                 web3.Address
+	ContractGasMetrics              *BlockGasMetrics
```

Comment:
Why not extract these values into `ContractMetrics`?

Reply:
Moved into dedicated struct 3c8fa5a904c5428e790ac7668be99c7f213e07ef